### DISCUSSION TITLE
how we can access the values provided to notes to a hosts

### DISCUSSION THREAD

**Discussion by lexjwa** [deployphp/deployer#2826] [Help needed]
We're looping and chaining up hosts for application servers.
 
It works fine if we knew what label we would like to use when deploying to application servers.

For example

```
dep deploy deployment_type=production
``` 

But we can have other types such as `staging`, `dev` or `production` for  label  `deployment_type`
 
 ```
$environment = 'Production'; // @todo this needs to be the value that was passed in as the deployment_type, first letter capitalized to match enum definition
$environmentValue = EnvironmentType::fromKey($environment)->value;
$servers = Server::active()->where('environment', $environmentValue)->with('instances')->get();
foreach($servers as $server) {
    $hostname = $server->uses_private_network ? $server->private_ip : $server->public_ip;
    foreach($server->instances as $instance) {
        if($instance->isActive()) {
            $deployPath = $instance->deploy_path;
            $instanceType = $instance->isWorker() ? 'worker' : 'application';
            $deploymentType = strtolower($server->environment->key); // "dev"|"staging"|"production"
            host($server->name)
                ->set('remote_user', 'remote_user_name')
                ->set('hostname', $hostname)
                ->set('deploy_path', $deployPath)
                ->set('keep_releases', 10)
                ->set('labels', ['deployment_type' => $deploymentType, 'server_type' => $instanceType]);
        }
    }
}
 ```
 
 Is there a way that we could know what value was provided to _label deployment_type_ in loops, so command to deploy is easy to use, perhaps something to list hosts and get what those hosts got in.

**Comment by antonmedv** [comment]
For example you can use env:

TYPE=Prod dep deploy all

and get env via getenv();

**Reply by nathan-io** [reply]
@antonmedv 

Thanks for this answer, but I'm a bit confused by it. The question is, when we run:

```
vendor/bin/dep deploy deployment_type=production
```

Is it possible to get the value that was provided for `deployment_type`?

Or are you saying instead of using the `deployment_type` label at all, we should pass "Prod" (or whatever) as a `TYPE` PHP environment variable, and fetch its value instead?

**Reply by antonmedv** [reply]
Something like this I suppose. But looks like people trying to use selector as input params. Maybe it should be allowed. 

Selector right now is not an input params. Consider next selectors:

- dep deploy all
- dep deploy role!=prod 
- dep deploy stage=prod and region!=us-east1
